src/builtins.ts

```diff
@@ -113,11 +114,24 @@ function any(x: collections.Iterable): boolean {
 }
 
 function bool(x) {
-    if (x === undefined)
+    
```

Comment:
It would be much cleaner and faster if all Python types had a `asBool` function that returns a boolean.

Then, this function would be:

```
function bool(x): boolean {
  if (typeof(x) === 'object') {
    // SingletonClass would need an `asBool`, too. Covers the None case.
    return x.asBool();
  } else {
    // Question: Is 'x' supposed to be a boolean, a number, or something else? If it's *any* primitive type, !!x will coerce it into an appropriate boolean value, but is not needed if x is known to be a boolean.
    return !!x;
  }
}
```